File: src/service/customer.service.js
import axios from "axios";

const API_URL= "http://localhost:9090";

class CustomerService{
    saveCustomer(customer) {
        return axios.post(API_URL+"/saveCustomer",customer);
    }

    getAllCustomer() {
        return axios.get(API_URL+"/");
    }

    getCustomerById(id) {
        return axios.get(API_URL+"/" + id);
    }

    deleteCustomer(id) {
        return axios.delete(API_URL+"/deleteCustomer/"+id);
    }

    updateCustomer(customer) {
        return axios.put(API_URL+"/updateCustomer/" + customer.id,customer);
    }
}

export default new CustomerService;

File: backend/src/main/java/com/example/restaurant/demo/restaurant/service/CustomerService.java
package com.example.restaurant.demo.restaurant.service;

import com.example.restaurant.demo.restaurant.model.CustomerReq;

import java.util.List;

public interface CustomerService {

    public CustomerReq saveCustomer(CustomerReq customer);

    public List<CustomerReq> getAllCustomer();
    public CustomerReq getCustomerById(Integer id);
    public String deleteCustomer(Integer id);
    public CustomerReq updateCustomer(CustomerReq customer,Integer id);


}
